File: 1768.merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i = j = 0
        merged = []

        # Try building result from both strings alternately. Check everytime
        # before adding from a string if has been exhausted. When both are
        # exhausted, we are done.
        while i < len(word1) or j < len(word2):
            merged.append(word1[i] if i < len(word1) else '')
            i += 1
            merged.append(word2[j] if j < len(word2) else '')
            j += 1

        return ''.join(merged)

File: 1768.merge-strings-alternately.c
char * mergeAlternately(char * word1, char * word2){
    int i, s1, j, s2, k;
    s1 = strlen(word1);
    s2 = strlen(word2);

    char* merged = (char*) malloc((s1+s2+1)*sizeof(char));
    i = j = k = 0;

    while (i < s1 || j < s2) {
        if (i < s1)
            merged[k++] = word1[i++];
        if (j < s2)
            merged[k++] = word2[j++];
    }

    merged[k] = '\0';
    return merged;
}
